apps/desktop-wallet/src/api/apiDataHooks/wallet/useFetchLatestTransactionOfEachAddress.ts

```diff
@@ -16,26 +16,24 @@ You should have received a copy of the GNU Lesser General Public License
 along with the library. If not, see <http://www.gnu.org/licenses/>.
 */
 
-import { useMemo } from 'react'
+import { useQueries } from '@tanstack/react-query'
 
 import { SkipProp } from '@/api/apiDataHooks/apiDataHooksTypes'
-import useFetchWalletSortedActivityTimestamps from '@/api/apiDataHooks/wallet/useFetchWalletSortedActivityTimestamps'
+import { flatMapCombine } from '@/api/apiDataHooks/apiDataHooksUtils'
+import { addressLatestTransactionQuery } from '@/api/queries/transactionQueries'
 import { useAppSelector } from '@/hooks/redux'
-import { selectAllAddressHashes } from '@/storage/addresses/addressesSelectors'
-
-const useFetchWalletAddressesHashesSortedByActivity = (props?: SkipProp) => {
-  const allAddressHashes = useAppSelector(selectAllAddressHashes)
-  const { data: addressesActivityTimestamps, isLoading } = useFetchWalletSortedActivityTimestamps(props)
-
-  const sortedAddressHashes = useMemo(
-    () => addressesActivityTimestamps.map(({ addressHash }) => addressHash),
-    [addressesActivityTimestamps]
-  )
-
-  return {
-    data: !isLoading && !props?.skip ? sortedAddressHashes : allAddressHashes,
-    isLoading
-  }
+import { useUnsortedAddressesHashes } from '@/hooks/useAddresses'
+
+const useFetchLatestTransactionOfEachAddress = (props?: SkipProp) => {
```

Comment:
I chose this name to not be confused with the (now deleted) `useFetchWalletLastTransactions` and `useFetchWalletTransactions[Limited|Infinite]`, since it doesn't return the _latest transactions of the wallet_ (which can potentially all be of the same address), but instead it returns the _latest tx of each address_.